### PR TITLE
Fix camp coordinates

### LINKED ISSUE #34438: Faction camps in buildings don't work unless building faces northward
# Describe the bug

When testing the fire station and evac shelter faction camp bases, I discovered they are not recognized as camps if they aren't north facing.

# Steps To Reproduce

1. Start a faction camp in an evac shelter or fire station that doesn't face north.
2. Try to access the bulletin board once it is placed.

# Expected behavior

camp should work in any rotation.

# Screenshots
image

# Versions and configuration

 - OS: [e.g. iOS 8 or Windows 10 or Ubuntu 18.04]
 - Game Version: [from the main menu, e.g. 0.C-29938-g90f5268437]
 - Graphics version: [Tiles or Terminal]
 - Mods loaded: [e.g.dda, boats, hacktheplanet, StatsThroughSkills]

# Additional context

- OS: Windows
    - OS Version: 10.0 1903
- Game Version: 0.D-8176-ga35bb25 [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Filthy Clothing [filthy_morale],
    Disable NPC Needs [no_npc_food]
]

### PR DESCRIPTION
**Pull request by AlexMooney**
#### Summary

```SUMMARY: Bugfixes "Can create camps in buildings that don't face North"```

#### Purpose of change

Fixes #34438 - Faction camps in buildings don't work unless building faces northward
Partial #34566 - faction camp not placing camp on shelter and teleporting NPC

The NPC still teleports away when asking them to join the camp, but now the camp is correctly created.

#### Describe the solution

Don't attempt to recompute `omt_pos` using a different method partway through camp creation.

#### Describe alternatives you've considered

I don't fully understand this code, but I thought that passing around the `&npc` exclusively to extract the `omt_pos` was a code smell.  It ended up being the source of the basecamp location bug.

It seems that the `&npc` is passed around in a lot of places throughout this part of the code.  If anyone feels strongly about leaving the signatures alone, I could instead only fix src/map.cppL5650.

#### Testing

- Tested building a camp in a south-facing shelter
- Tested building a camp in a West-facing shelter

### CODE REVIEW

File: src/basecamp.cpp [hunk 1]
```diff
@@ -164,10 +164,9 @@ void basecamp::add_expansion( const std::string &bldg, const tripoint &new_pos,
     update_resources( bldg );
 }
 
-void basecamp::define_camp( npc &p, const std::string &camp_type )
+void basecamp::define_camp( const tripoint omt_pos, const std::string &camp_type )
 {
     query_new_name();
-    omt_pos = p.global_omt_location();
```
Comment (Qrox) [inline src/basecamp.cpp]: `omt_pos` here is actually a member of `basecamp`, so you should still assign the argument value to it.

Comment (AlexMooney) [reply src/basecamp.cpp]: Ah gotcha.  Could I do something like `self.omt_pos = omt_pos` or should I rename the argument in the signature variable and do `omt_pos = p` or whatever?

Comment (mlangsdorf) [reply src/basecamp.cpp]: the latter.

**Comment by meelock** [comment]
This is not working for me, seems this pr was broken recently.

**Comment by AlexMooney** [comment]
@meelock could you either comment on one of the issues or else open a new one, please?  We need more details to hunt down the regression.

**Pull request merged**